theano/gpuarray/opt.py

```diff
@@ -947,16 +1285,19 @@ def local_inplace_sparseblockouter(node):
 @op_lifter([AbstractConv2d,
             AbstractConv2d_gradWeights,
             AbstractConv2d_gradInputs])
-def local_lift_abstractconv2d(node, context_name):
-    if isinstance(node.outputs[0].type, GpuArrayType):
+@register_opt2([AbstractConv2d,
+                AbstractConv2d_gradWeights,
+                AbstractConv2d_gradInputs], 'fast_compile')
+def local_lift_abstractconv2d(op, context_name, inputs, outputs):
+    if isinstance(inputs[0].type, GpuArrayType):
         # Don't handle this node here, it's already on the GPU.
```

Comment:
I don't think that function needs to be decorated with @register_opt2, but I'm not sure, and we may need it in the future.
For the moment, can you correct that comment (the node is not necessarily on the GPU), saying that if called from the new optimizer (gpuarray_graph_optimization), the optimization will be skipped.